src/wsdd.py

```diff
@@ -779,7 +779,25 @@ def serve_wsd_requests(addresses):
                 http.server.HTTPServer
                 if interface.family == socket.AF_INET
                 else HTTPv6Server)
-            http_srv = klass(interface.listen_address, WSDHttpRequestHandler)
+            logger.info('Interface address: {}'
+                        .format(interface.listen_address[0]))
+            retries = 0
+            while True:
+                try:
+                    http_srv = klass(interface.listen_address,
+                                     WSDHttpRequestHandler)
+                    break
+                except OSError:
+                    logger.warning('Http server start failed, retrying, \
+                                    interface address: {}'
+                                   .format(interface.listen_address[0]))
+                    retries += 1
+                    if retries > 10:
+                        logger.error('Http server start failed, exiting, \
+                                     interface address: {}'
+                                     .format(interface.listen_address[0]))
+                        return 0
+                    time.sleep(1)
```

Comment:
If one adds `sleep 10` to the unit file, e.g., then this would do basically the same as the loop here: bind to the interface and if it works we're fine; and otherwise we terminate due an OS error. The intention of the loop, I think, is to wait until one can bind the IPv6 address, i.e. the address is not tentative anymore. In other words, let's wait for the DAD procedure to be completed. The only advantage I see at the moment is that the loop terminates earlier than a simple sleep for 10 seconds if the interfaces address becomes usable. For the user, however, this would not really make a hard difference if the freshly booted system will become visible a few seconds earlier or not. 

Overall, simply waiting for given amount of time is not a satisfying solution at all as DAD may still (or again) be in process.